src/main/java/Duke/CommandList.java

```diff
@@ -0,0 +1,133 @@
+
+            if (firstWord.equals("mark")) {
+                Task targetTask = list.get((index - 1));
+                targetTask.mark();
+            } else {
+                Task targetTask = list.get((index - 1));
+                targetTask.unmark();
+            }
+            storage.store(list);
+        } else if (firstWord.equals("delete")) { // delete task
+            String secondWord = inputs[1];
+            int index = Integer.parseInt(secondWord);
+
+            Task targetTask = list.get((index - 1));
+            list = targetTask.delete(list, (index - 1));
+            System.out.println("Now you have " + list.size() + " tasks in your list.");
+            storage.store(list);
+        } else if (firstWord.equals("todo")) { // adds a todo task
+            if (inputs.length == 1) {
+                System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+                parser.terminate();
+            }
+            ToDo toDo = new ToDo(input.substring((5)));
+            list.add(toDo);
+            System.out.println("Got it. I've added this task:\n"
+                    + toDo + "\nNow you have " + list.size() + " tasks in your list.");
+            storage.store(list);
+        } else if (firstWord.equals("deadline") || firstWord.equals("event")) {
+            int start = input.indexOf("/");
+            if (start == -1) { // if / char cannot be found
+                System.out.println("Please enter a valid date!");
+            }
+            String date = input.substring(start + 4);
+            if (firstWord.equals("deadline")) { // adds a deadline task
+                if (inputs.length == 1) {
+                    System.out.println("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    parser.terminate();
+                }
+                try {
+                    SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+                    Date formattedDate = dateFormatter.parse(date);
+                    String dateToString = new SimpleDateFormat("MMM-dd-yyyy HHmm").format(formattedDate);
+
+                    Deadline deadline = new Deadline(input.substring(9, start - 1), dateToString);
+                    list.add(deadline);
+                    System.out.println("Got it. I've added this task:\n"
+                            + deadline + "\nNow you have " + list.size() + " tasks in your list.");
+                    storage.store(list);
+                } catch (ParseException e) {
+                    System.out.println("Please enter a valid date!");
+                }
+            } else { // adds an event task
+                if (inputs.length == 1) {
+                    System.out.println("☹ OOPS!!! The description of an event cannot be empty.");
+                    parser.terminate();
+                }
+                try {
+                    SimpleDateFormat dateFormatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
```

Comment:
There's some copy-pasted code here with parsing and formatting the date. Maybe you could extract this out into a function to avoid code duplication?